clippy_lints/src/slow_vector_initialization.rs

```diff
@@ -32,13 +46,17 @@ declare_clippy_lint! {
     ///
     /// let mut vec2 = Vec::with_capacity(len);
     /// vec2.extend(repeat(0).take(len));
+    ///
+    /// let mut vec3 = Vec::new();
```

Comment:
I would make this the second (maybe even the first?) example (and maybe cut the `vec1` example, since it seems fairly obviously similar to the `vec0` example). In contrast, this last example is already more effcient than the other ones because it only allocates once.

Reply:
Updated the examples, is this what you had in mind?